### PR TITLE
Add a idstore-export CLI tool

### LINKED ISSUE #111: Credential migration for genesis
We want to migrate the pre-alpha credentials to the alpha network for genesis. 

Write a tool to dump the credentials from the pre-alpha merk storage and import them into the alpha merk storage.

### PR DESCRIPTION
**Pull request by hansl**
Fixes #111

### CODE REVIEW

File: src/idstore-export/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        let new_v = Tree::decode(key.to_vec(), value.as_ref());
+        let value = new_v.value().to_vec();
+
+        idstore.insert(base64::encode(key.as_ref()), base64::encode(&value));
+    }
+
+    let root = JsonRoot {
+        seed: merk
+            .get(b"/config/idstore_seed")
+            .expect("Could not read see")
```
Comment (fmorency) [inline src/idstore-export/src/main.rs:46]: seed

File: src/many-ledger/src/module.rs [hunk 2]
```diff
@@ -137,18 +137,33 @@ impl LedgerModuleImpl {
...
                     account
                         .create_account(&mut storage)
                         .expect("Could not create accounts");
                 }
-                storage.commit_persistent_store().expect("Could not commit");
+                should_commit = true;
             }
 
+            if should_commit {
+                storage.commit_persistent_store().expect("Could not commit");
```
Comment (fmorency) [inline src/many-ledger/src/module.rs:165]: Not sure I understand why you did this?

File: src/many-ledger/src/storage.rs [hunk 3]
```diff
@@ -455,20 +460,28 @@ impl LedgerStorage {
...
+        let mut idstore_seed = self
+            .persistent_store
+            .get(b"/config/idstore_seed")
+            .unwrap()
+            .map_or(0u64, |x| {
+                let mut bytes = [0u8; 8];
+                bytes.copy_from_slice(x.as_slice());
+                u64::from_be_bytes(bytes)
+            });
+        idstore_seed += 1;
```
Comment (fmorency) [inline src/many-ledger/src/storage.rs:472]: The first seed should be `0`, not `1`

**Review by fmorency — CHANGES_REQUESTED** [review]
Thanks! Some comments!

Can you add a test? We want to make sure this works as expected!

Also, fix CI! The tests don't build at the moment.

**Review by hansl — COMMENTED** [review]
Not sure how to make a test, I'll wait for your bats WebAuthn tests before I can make one.

**Comment by hansl** [comment]
@fmorency Added a test, PTAL.

File: tests/e2e/webauthn.bats [hunk 4]
```diff
@@ -18,44 +18,80 @@ function teardown() {
...
+
+    cat $import_file >&2
+
+    start_ledger \
+        --persistent="$ledger_db" \
+        --state="$import_file" \
+        --pem "$(pem 0)" \
+        --disable-webauthn-only-for-testing # Disable WebAuthn check for this test
+
+    # Continue the test.
```
Comment (fmorency) [inline tests/e2e/webauthn.bats:86]: You're storing the same identity/key twice. 

To be on the safe side, you should generate and store a new one. Test that you can retrieve the old one as well as the new one using the recall phrase and address.

**Review by fmorency — CHANGES_REQUESTED** [review]
A small change in testing logic is required. Else LGTM.

**Review by fmorency — APPROVED** [review]

**Pull request merged**